dev/com.ibm.ws.artifact.zip/src/com/ibm/ws/artifact/zip/internal/ZipFileContainerUtils.java

```diff
@@ -453,36 +454,170 @@ public int getOffset() {
         private final String r_path;
 
         @Trivial
+        @Override
         public String getPath() {
             return path;
         }            
 
         @Trivial
+        @Override
         public boolean isDirectory() {
-            return ( path.charAt( path.length() - 1 ) == '/' ); 
+            return false; 
         }
 
         @Trivial
+        @Override
         public String r_getPath() {
             return r_path;
         }
 
         //
         
-        private final long size;
+        private final int size;
         private final long time;
 
         @Trivial
-        public long getSize() {
+        @Override
+        public int getSize() {
             return size;
         }
         
         @Trivial
+        @Override
         public long getTime() {
             return time;
         }
     }
     
+    private static class DirZipEntryData implements ZipEntryData {
```

Comment:
I'm wondering if you could save some code by using inheritance here...  not a show-stopper though.

Reply:
I added some inheritance to address your review comments.